lib/client/result-client/index.js

```diff
@@ -0,0 +1,224 @@
+
+  var donePromise = mediator.promise(topicGenerators.syncDoneTopic({topicUid: topicUid, topicName: CONSTANTS.TOPICS.CREATE}));
+
+  var errorPromise = mediator.promise(topicGenerators.syncErrorTopic({topicUid: topicUid, topicName: CONSTANTS.TOPICS.CREATE}));
+
+  mediator.publish(topicGenerators.syncTopic({topicName: CONSTANTS.TOPICS.CREATE}), topicParams);
+
+  return getTopicPromises(donePromise, errorPromise);
+}
+
+/**
+ *
+ * Updating an existing result.
+ *
+ * @param {object} resultToUpdate - The result to update
+ * @param {string} resultToUpdate.id - The ID of the result to update
+ */
+function update(resultToUpdate) {
+
+  console.log("Result Update", resultToUpdate);
+  var topicUid = resultToUpdate.id || resultToUpdate._localuid;
+
+  var topicParams = {topicUid: topicUid, itemToUpdate: resultToUpdate};
+
+  var donePromise = mediator.promise(topicGenerators.syncDoneTopic({topicUid: topicUid, topicName: CONSTANTS.TOPICS.UPDATE}));
+
+  var errorPromise = mediator.promise(topicGenerators.syncErrorTopic({topicUid: topicUid, topicName: CONSTANTS.TOPICS.UPDATE}));
+
+  mediator.publish(topicGenerators.syncTopic({topicName: CONSTANTS.TOPICS.UPDATE}), topicParams);
+
+  return getTopicPromises(donePromise, errorPromise);
+}
+
+/***
+ *
+ * Reading a single result.
+ *
+ * @param {string} resultId - The ID of the result to read
+ */
+function read(resultId) {
+
+  var topicGenerationParams = {topicName: CONSTANTS.TOPICS.READ};
+
+  var donePromise = mediator.promise(topicGenerators.syncDoneTopic({topicName: CONSTANTS.TOPICS.READ, topicUid: resultId}));
+  var errorPromise = mediator.promise(topicGenerators.syncErrorTopic({topicName: CONSTANTS.TOPICS.READ, topicUid: resultId}));
+
+  mediator.publish(topicGenerators.syncTopic(topicGenerationParams), {id: resultId, topicUid: resultId});
+
+  return getTopicPromises(donePromise, errorPromise);
+}
+
+/**
+ * Listing All Results
+ */
+function list() {
+  var topicParams = {topicName: CONSTANTS.TOPICS.LIST};
+
+  var donePromise = mediator.promise(topicGenerators.syncDoneTopic(topicParams));
+  var errorPromise = mediator.promise(topicGenerators.syncErrorTopic(topicParams));
+
+  mediator.publish(topicGenerators.syncTopic(topicParams));
+
+  return getTopicPromises(donePromise, errorPromise);
+}
+
+/**
+ *
+ * Removing a workororder using the sync topics
+ *
+ * @param {object} resultToRemove
+ * @param {string} resultToRemove.id - The ID of the workoroder to remove
+ */
+function remove(resultToRemove) {
+
+  var donePromise = mediator.promise(topicGenerators.syncDoneTopic({topicName: CONSTANTS.TOPICS.REMOVE, topicUid: resultToRemove.id}));
+  var errorPromise = mediator.promise(topicGenerators.syncErrorTopic({topicName: CONSTANTS.TOPICS.REMOVE, topicUid: resultToRemove.id}));
+
+  mediator.publish(topicGenerators.syncTopic({topicName: CONSTANTS.TOPICS.REMOVE}), {id: resultToRemove.id, topicUid: resultToRemove.id});
+
+  return getTopicPromises(donePromise, errorPromise);
```

Comment:
function returns but it isn't added to function comment. All functions returning `getTopicPromises()` are missing that part of the jsdoc